### ISSUE TITLE
Resize elastic volume service disk destorys and recreates disk

### ISSUE THREAD

**Issue by m3bg** [opentelekomcloud/terraform-provider-opentelekomcloud#656]
Hi there,

resizeing an evs disk destroys and recreates the disk.

# Terraform Version
Terraform v0.12.29
+ provider.opentelekomcloud v1.20.3

# Affected Resource(s)
Please list the resources as a list, for example:
- opentelekomcloud_blockstorage_volume_v2
# Terraform Configuration Files
Applyed configuration baseline:
```hcl
resource "opentelekomcloud_blockstorage_volume_v2" "evs-dev-var" {
  name              = "evs-playground-dev-var"
  size              = "50"
  volume_type       = "SATA"
  device_type       = "VBD"
  availability_zone = "eu-de-03"
  metadata = {
    policy = ""
    attached_mode = "rw"
    readonly      = "False"
  }
```
Changed to:
```hcl
resource "opentelekomcloud_blockstorage_volume_v2" "evs-dev-var" {
  name              = "evs-playground-dev-var"
  size              = "60"
  volume_type       = "SATA"
  device_type       = "VBD"
  availability_zone = "eu-de-03"
  metadata = {
    policy = ""
    attached_mode = "rw"
    readonly      = "False"
  }
```

# Debug Output/Panic Output
```plain
Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value:
```
# Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
```bash
terraform apply
```

# Expected Behavior
The evs disk should be resized. It should behave like if the size is changed in the OTC gui. 

# Actual Behavior
The disk is destroyed and recreated.

Rgds
Marcel

**Comment by outcatcher** [comment]
@m3bg

From the API point of view, only [expanding](https://docs.otc.t-systems.com/en-us/api/evs/evs_04_2083.html) volume capacity can be done.
For decreasing volume size, you have to destroy volume and create it again.

The possibility to update the resource if it's attribute has changed is defined by the `ForceNew` property.

**So there are several possible solutions:**
1. Re-create volume on size change (as it works now)
2. Return an error if volume size is decreased and update correctly if the size is increased 
3. *(Looks weird, but don't break anything)* Add resource attribute `extend_size` which describes the size of additional storage to be added. Changing size attribute will still lead to re-creation. Decreasing the `extend_size` value will lead to an exception.

Openstack provider has the same `ForceNew` [approach](https://github.com/terraform-provider-openstack/terraform-provider-openstack/blob/4ad9f720fdf41ff99e2747dcf34bb3eb41ba560a/openstack/resource_openstack_blockstorage_volume_v2.go#L40)

**Comment by m3bg** [comment]
@outcatcher 
@sfhl

I think 2. is the most obvious way!?

Are there any users out there witch use the destroy/recreate on resize as a feature?

Rgds
Marcel

**Comment by outcatcher** [comment]
@m3bg The problem with the second option is breaking backward compatibility. I understand that it's hardly a feature, but the need to destroy resources manually can be a problem for some users.

I will pin this issue :) let's wait for a week if someone will have comments for that. 
Is it okay for you?

**Comment by m3bg** [comment]
@outcatcher YES

**Comment by outcatcher** [comment]
Sorry for misleading you. After some more investigation, there is a way to use optional `ForceNew`.
Now I will implement the re-creation of the resource only on decreasing size.

**Comment by m3bg** [comment]
@outcatcher :thumbsup: Thanks!

**Comment by outcatcher** [comment]
Fixed in v1.21.1: destroy volume only on size decreasing
